Tests/WebSocketKitTests/AsyncWebSocketKitTests.swift

```diff
@@ -42,10 +43,56 @@ final class AsyncWebSocketKitTests: XCTestCase {
         try await server.close(mode: .all)
     }
 
+    func testWebSocketSendCodable() async throws {
+        let server = try ServerBootstrap.webSocket(on: self.elg, logger: self.logger) { req, ws in
+            ws.onEvent("hello", User.self) { ws, user in
+                ws.send("Hello \(user.firstName) \(user.lastName)")
+            }
+        }.bind(host: "localhost", port: 0).wait()
+
+        guard let port = server.localAddress?.port else {
+            XCTFail("couldn't get port from \(server.localAddress.debugDescription)")
+            return
+        }
+
+        let promise = elg.next().makePromise(of: String.self)
+
+        try await WebSocket.connect(to: "ws://localhost:\(port)", on: elg) { ws in
+            do {
+                try await ws.send(Response(event: "hello", data: User(firstName: "Vapor", lastName: "WebSocket")))
+                ws.onText { ws, string in
+                    promise.succeed(string)
```

Comment:
We shouldn't be touching promises in async tests

Reply:
I get the template of that test from the already existing test above it.  Btw changed it...